ext/src/content/content.js

```diff
@@ -268,50 +277,95 @@ function init() {
+            const codeSnippetRegex = /(`{1,2})([^\n`]+)(`{1,2})/gu;
             const [firstLine] = lines;
-            const [, lang = null] =
+
+            if (langAllRegex.test(firstLine)) {
+              [, langAll = null] = firstLine.match(langAllRegex);
+              return;
+            }
+
+            const [, lang = langAll] =
               firstLine.match(codeBlockRegex) ||
-              firstLine.match(indentedCodeRegex) ||
+              firstLine.match(indentedCodeWithLangRegex) ||
               [];
 
             if (!lang) {
               return;
             }
 
-            const isCodeBlock = codeBlockRegex.test(firstLine);
-            const indentedLineCodeBlockStartIdx = 2;
-            const codeLines = isCodeBlock
-              ? lines.slice(1, -1)
-              : lines.slice(indentedLineCodeBlockStartIdx);
-            let formattedBlock = window.prettier.format(codeLines.join("\n"), {
-              parser: PARSERS_LANG_MAP[lang],
-              plugins: window.prettierPlugins
-            });
-
-            // Prettier adds a trailing newline
-            if (codeLines.length !== formattedBlock.split("\n").length) {
-              formattedBlock = formattedBlock.replace(/\n$/u, "");
-            }
+            let formattedText = lines.join("\n");
+
+            // Code Snippets
+            if (codeSnippetRegex.test(firstLine)) {
+              formattedText = firstLine.replace(
+                codeSnippetRegex,
+                (match, openingBackticks, snippet, closingBackticks) => {
+                  let formattedSnippet = snippet;
+
+                  try {
+                    formattedSnippet = window.prettier.format(snippet, {
+                      parser: PARSERS_LANG_MAP[lang],
+                      plugins: window.prettierPlugins
+                    });
+                  } catch {}
 
-            formattedBlock = isCodeBlock
-              ? `${firstLine}\n${formattedBlock}\n${lines[lines.length - 1]}`
-              : `${firstLine}\n${lines[1]}\n    ${
-                  formattedBlock.split("\n").length > 1
-                    ? formattedBlock.split("\n").join("\n    ")
-                    : formattedBlock
+                  return `${openingBackticks}${formattedSnippet}${closingBackticks}`;
+                }
+              );
+
+              // Code Blocks
+            } else {
+              const isCodeBlock = codeBlockRegex.test(firstLine);
+              const isIndentedBlockWithLang = indentedCodeWithLangRegex.test(
+                firstLine
+              );
+              let codeLines;
+
+              if (isCodeBlock) {
+                codeLines = codeLines.slice(1, -1);
+              } else {
+                const indentedLineCodeBlockStartIdx = 2;
+                codeLines = isIndentedBlockWithLang
+                  ? lines.slice(indentedLineCodeBlockStartIdx)
+                  : lines;
+              }
+
+              try {
```

Comment:
Wrapping this in a `try`/`catch` statement allows the plugin to continue onto the next code snippet or block without crashing. This will probably happen frequently since Prettier throws if the parser can't parse something.